pkg/volume/volume.go

```diff
@@ -0,0 +1,192 @@
+package volume
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	adminv1 "github.com/acorn-io/acorn/pkg/apis/internal.admin.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	"github.com/acorn-io/baaah/pkg/typed"
+	"golang.org/x/exp/slices"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubectl/pkg/util/storage"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func SyncVolumeClasses(req router.Request, resp router.Response) error {
```

Comment:
Why are we syncing all StorageClasses during every reconciliation? The request should have the specific StorageClass and we can just apply that one instead of syncing all of them, right?

`sc := req.Object.(*storage.StorageClass)`

Basically something like this.

Reply:
I am doing it this way so that the `ephemeral` class gets created as expected. If we only handled the storage class in the request, then we would never create the ephemeral one.

However, you question raises the point: what happens if there are no storage classes? Then the ephemeral class will never get created. I will fix this.